docs/DeveloperGuide.md

```diff
@@ -1,34 +1,614 @@
+commands and as well as to allowing HAMS to exit.
+
+![](images/Command%20class%20diagram.png)
+
+All of these command classes inherits from the abstract ```Command``` class. Likewise, the execute function of each 
+command  class is also inherited from the abstract ```Command``` class's ```execute(Ui ui, Storage storage)``` function 
+too. Every command class (other than the  ```ExitCommand``` and ```HelpCommand```) are actually façade classes that 
+creates the connection from the ```Main``` class to the other classes required such as ```Storage``` class,
+```PatientList``` class and the ```Appointment``` class to name a few. 
+
+##### 2.2.4.1 AddPatientClass
+
+To add a patient, the ```AddPatientCommand``` class is used. For this ```AddPatientCommand``` class, it serves as the 
+façade class for the ```Main```, ```Patient``` , ```PatientList``` and the ```Storage``` class to interact with one 
+another. Also, to uniquely identify a patient, an unique patientId number is assigned to each patient when they are first added into the patient list.
+
+![](images/AddPatientDiagram.png)
+
+1. The ```AddPatientCommand``` class object will first be created by the ```Parser``` object, where the information 
+regarding the patient to be added will be stored in a Map, where the ```AddPatientCommand``` class object would read the Map content and store the information about the patient in said ```AddPatientCommand``` class object. 
+For the patient Id number, it will call upon the static class ```patientIdManager``` to get its unique patient id number. This unique Id number will be used later in the ```Patient``` object creation too.
+
+2. When the 
+```execute(Ui ui, Storage storage)``` command is called, the  ```AddPatientCommand``` would first make use of the 
+```Patient``` class constructor to create a new ```Patient``` object based on the information stored back in step 1. 
+
+3. After which, it would then call the 
+```PatientList```’s ```getPatientList()``` command to get the ```List``` patient list object such that the ```Patient``` 
+object created beforehand can directly be inserted into the patient list. 
+
+4. After adding the patient into the patient list 
+object, the ```Storage```’s ```savePatientList()``` function will be called next so that the newly update list of 
+```patient``` is saved as offline data.  
+
+5. When the above operation is successful, it will call upon the ```Ui``` class’ 
+```showPatientAddSuccess()``` function to display the success of adding the ```Patient``` object into the patient list.
+
+If the supplied patient age is a word or is missing, the age will be set to ```-1```. This value is chosen to indicate 
+that there isn’t a valid age set. Thus, when displaying the age, if ```-1``` is encountered, show age as an empty string 
+instead.
+
+Below shows the sequence diagram for ```AddPatientCommand``` class
+
+![](images/AddPatientCommandSequence.png)
```

Comment:
![image](https://user-images.githubusercontent.com/49268567/77992796-5f842200-7359-11ea-8fa5-cf8d183ffc64.png)
When geting the record, do you mean calling the method `getPatientRecord` of the patient class? If you are calling the method from the storage class, maybe the arrow should points to the storage.